plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -115,93 +75,36 @@ func TestWrite(t *testing.T) {
 
 			localFakeIngestor := &fakeIngestor{}
 			plugin := AzureDataExplorer{
-				Endpoint:        "someendpoint",
-				Database:        "databasename",
-				Log:             testutil.Logger{},
-				MetricsGrouping: tC.metricsGrouping,
-				TableName:       tC.tableName,
-				CreateTables:    tC.createTables,
-				kustoClient:     kusto.NewMockClient(),
-				metricIngestors: map[string]ingest.Ingestor{
-					tC.tableName: localFakeIngestor,
+				Config: common_adx.Config{
+					Endpoint:        "https://someendpoint.kusto.net",
+					Database:        "databasename",
+					MetricsGrouping: tC.metricsGrouping,
+					TableName:       tC.tableName,
+					CreateTables:    tC.createTables,
+					IngestionType:   ingestionType,
 				},
-				serializer:    serializer,
-				IngestionType: ingestionType,
+				serializer: serializer,
+				Log:        testutil.Logger{},
 			}
-
+			plugin.client = NewMockClient(&plugin.Config, plugin.Log, map[string]ingest.Ingestor{
+				tC.tableName: localFakeIngestor,
+			})
```

Comment:
There is no value in this test I think. You could do the same in `common/adx` (and you already do), so why do it here introducing the need for interfaces etc...

Reply:
It helps UTing the Writr() flow by mocking the PushMetric()

Reply:
Sorry I'm still not getting it. You have different test-cases but you are only testing if write does not produce an error. How is that helpful? Imaging I delete all logic in the plugin to distinguish between table per metric and single table writes. This test would just pass. So what is the benefit? I could understand if you check the results or check for certain commands issued to the client but in this form the test is useless IMO.